service/src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -362,24 +363,75 @@ public void createReferenceResource(ReferencedResource resource)
     storeResource(resource);
   }
 
+  /**
+   * Update name, description, and/or attributes of the resource.
+   *
+   * @param name name of the resource, may be null if it does not need to be updated
+   * @param description description of the resource, may be null if it does not need to be updated
+   * @param attributes resource-type specific attributes, may be null if it does not need to be
+   *     updated .
+   */
   @WriteTransaction
   public boolean updateResource(
-      UUID workspaceId, UUID resourceId, String name, String description) {
-    if (name == null && description == null) {
+      UUID workspaceId,
+      UUID resourceId,
+      @Nullable String name,
+      @Nullable String description,
+      @Nullable String attributes) {
+    if (name == null && description == null && attributes == null) {
       return false;
     }
 
     var params = new MapSqlParameterSource();
 
-    if (name != null) {
+    if (!StringUtils.isEmpty(name)) {
       params.addValue("name", name);
     }
-
-    if (description != null) {
+    if (!StringUtils.isEmpty(description)) {
       params.addValue("description", description);
     }
+    if (!StringUtils.isEmpty(attributes)) {
+      params.addValue("attributes", attributes);
+    }
+
+    StringBuilder sb = new StringBuilder("UPDATE resource SET ");
+
+    if (params.hasValue("attributes")) {
+      sb.append(DbUtils.setColumnsClause(params, "attributes"));
+    } else {
+      sb.append(DbUtils.setColumnsClause(params));
+    }
+
+    sb.append(" WHERE workspace_id = :workspace_id AND resource_id = :resource_id");
+
+    MapSqlParameterSource queryParams = new MapSqlParameterSource();
+    queryParams
+        .addValues(params.getValues())
+        .addValue("workspace_id", workspaceId.toString())
+        .addValue("resource_id", resourceId.toString());
 
-    return updateResourceColumns(workspaceId, resourceId, params);
+    int rowsAffected = jdbcTemplate.update(sb.toString(), queryParams);
+    boolean updated = rowsAffected > 0;
+
+    logger.info(
+        "{} record for resource {} in workspace {}",
+        (updated ? "Updated" : "No Update - did not find"),
+        resourceId,
+        workspaceId);
+
+    return updated;
+  }
+
+  /**
+   * Update name and/or description of the resource.
+   *
+   * @param name name of the resource, may be null if it does not need to be updated
+   * @param description description of the resource, may be null if it does not need to be updated
+   */
+  @WriteTransaction
+  public boolean updateResource(
+      UUID workspaceId, UUID resourceId, @Nullable String name, @Nullable String description) {
+    return updateResource(workspaceId, resourceId, name, description, null);
```

Comment:
This is a form I would like us to avoid: where one public method with a transaction annotation calls another public method with a transaction annotation. It is not technically wrong. It is so we always know where transactions are getting started. It mostly helps in debugging.

The fix would be to make a `updateResourceWorker` method that is a private method. It would take the same param list as the first `updateResource`. Then the two public methods would call the worker. One with attributes and the other with null. Does that make sense?